packages/astro-google-fonts-optimizer/font-utils.ts

```diff
@@ -11,20 +13,35 @@ const userAgents: { name: string; ua: string }[] = [
   },
 ];
 
-export function downloadFontCSS(url: string): Promise<string> {
-  const fontDownloads = Promise.all(
-    userAgents.map((entry) => {
-      return fetch(url, { headers: { 'User-Agent': entry.ua } })
-        .then((res) => {
-          if (!res.ok) {
-            throw new Error(res.statusText);
-          }
-          return res.text();
-        })
-        .then((t) =>
-          t.replace(/  +/g, '').replace(/\t+/g, '').replace(/\n+/g, '')
-        );
+const fetchFunc = <Args extends (typeof userAgents)[number]>(
+  entry: Args,
+  url: string,
+) =>
+  fetch(url, { headers: { 'User-Agent': entry.ua } })
+    .then((res) => {
+      if (!res.ok) {
+        throw new Error(res.statusText);
+      }
+      return res.text();
     })
+    .then((t) => t.replace(/  +/g, '').replace(/\t+/g, '').replace(/\n+/g, ''));
+
+export const initialise = (
+  retryAttempts: number = 3,
+  retryInterval: number = 100,
+) => {
+  const cachedFetch = cacheWrapper(
```

Comment:
when I understand it correctly, the cacheWrapper has no effect when downloading the same font in another instance of the Astro component as intialise get's called in every component instance, so the cache is setup independently for every component. Am I wrong?

And also regarding caching in general: I originally meant with it that the downloaded resources get cached (somewhere), so when building the Astro project again, there's not always the need to download the resources again. But I also like the idea of preventing downloading the same font multiple times with a cache.

Reply:
You're right there

And since it's a in-memory cache, it resets for each astro build/launch

As I understand you, your idea is that cache will be persistent so font once downloaded will be used in dev and build?

Reply:
yes